File: IHaveATheory/app/src/main/java/com/avoid/ihaveatheory/game/Gather.java
package com.avoid.ihaveatheory.game;

public class Gather {
    private static final Gather instance = new Gather();

    private Gather() {

    }

    public static Gather getInstance() {
        return instance;
    }

    public int findWater() {
        return 5;
    }

    public int findBerries() {
        return 10;
    }

    public int findWood() {
        return 3;
    }

    public int findRabbits() {
        return 1;
    }
}

File: app/src/main/java/com/avoid/ihaveatheory/game/Gather.kt
package com.avoid.ihaveatheory.game

import org.jetbrains.annotations.Contract

class Gather {

    companion object {

        @get:Contract(pure = true)
        val instance = Gather()

        @JvmStatic
        fun findWater(): Int {
            return 5
        }

        @JvmStatic
        fun findBerries(): Int {
            return 10
        }

        @JvmStatic
        fun findWood(): Int {
            return 3
        }

        @JvmStatic
        fun findRabbits(): Int {
            return 1
        }

    }
}
